atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,187 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import logging
+import json
+import paramiko
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class Host:
+    '''
+    Remote host abstract class
+    '''
+    def __init__(self, username: str, hostname: str, arch: str):
+        self.username = username
+        self.hostname = hostname
+        self.arch = arch
+        self.slots = []
+        self.free_slots = 0
+
+    def _read_info_data(self):
+        cmd = f"cat {ROOT_PATH}info.json"
+        data = self._ssh_run_remote_command(cmd)
+        return data
+
+    def _read_slot_data(self, slot: int):
+        cmd = f"cat {ROOT_PATH}slot_{slot}.json"
+        data = self._ssh_run_remote_command(cmd)
+        return data
+
+    def check_avail(self):
+        try:
+            data = self._read_info_data()
+            if data == '':
+                logger.warning("%s hasn't been deployed yet", self.hostname)
+                return False
+            else:
+                # parse slot count, arch
+                meta = json.loads(data)
+                logger.debug("%s is up", meta['hostname'])
+                # check slots status
+                i = 0
+                while i < meta.get('max_slot_count', 1):
+                    slot = self._read_slot_data(i)
+                    if slot == b'0\n':
+                        state = 'free'
+                        prid = ''
+                        self.free_slots += 1
+                    elif slot == "":
+                        logger.warning("host: %s slot: %s has invalid slot state", self.hostname, i)
+                        state = 'unknown'
+                        prid = 'unknown'
+                    else:
+                        state = 'running'
+                        logger.debug(slot)
+                        slot_data = json.loads(slot)
+                        prid = slot_data.get("prid", "unknown")
+                    logger.debug("%s slot %s is %s", meta['hostname'], i, state)
+                    self.slots.append({'sid': i, 'state': state, 'prid': prid})
+                    i += 1
+                return True
+        except Exception as err:
+            logger.debug(err)
+            logger.error("Availiability check failed fo host: %s", self.hostname)
+            return False
+
+    def _write_free_slot(self, slot: int):
+        cmd = f'echo \'0\' > {ROOT_PATH}slot_{slot}.json'
```

Comment:
I see that we're using f-strings and format for strings. Would it be possible to unify this for whole PR and use f-strings everywhere?

Reply:
f strings are not accepted by pylint. W1203(logging-fstring-interpolation)

another solution is to use:
 cmd = 'echo \'0\' > {}slot_{}.json'.format(ROOT_PATH, slot)

what do you prefer?

Reply:
That's incorrect. The line in question is not logging.

We *do* (unfortunately) have two standards for strings: always single-quoted (except where simply it's not possible) and

1. f-interpolated
2. lazy eval (i.e. %s formatting) - ONLY for strings used in logging statements, as the logger itself doesn't understand f-strings - although that may have changed, but either way, lazy-eval is desirable for logging

If pylint is raising W1203 for `cmd = f'echo \'0\' > {ROOT_PATH}slot_{slot}.json'`, then there's a different issue at play (pylint bug, configuration issue...).

Reply:
I don't understand this comment. there is already f-string used and pylint doesn't complain

Reply:
So based on @ben-alkov comment - we cannot have f-strings everywhere as @lkolacek requested.

-------
yeah, I wasn't right. Thanks Ben for clarification